addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -4039,6 +4039,28 @@ QUnit.module("Views", (hooks) => {
         assert.containsNone(document.body, ".modal", "there should not be a confirm modal");
     });
 
+    QUnit.test("Domain: allow empty domain on fieldInfo", async function (assert) {
+        serverData.models.partner.fields.product_id.domain = "[('display_name', '=', name)]";
+        const form = await makeView({
+            type: "form",
+            resModel: "partner",
+            serverData,
+            arch: `
+                <form>
+                    <header>
+                        <field name="product_id" widget="statusbar" domain="[]"></field>
+                    </header>
+                    <sheet>
+                        <group>
+                            <field name="name"></field>
+                        </group>
+                    </sheet>
+                </form>`,
+            resId: 1,
+        });
+        assert.strictEqual(form.archInfo.activeFields.product_id.domain, "[]");
```

Comment:
introspection is discouraged, prefer triggering the name_get by clicking on the many2one